epam_tr/src/main/java/com/epam/tr/service/FileSystemService.java

```diff
@@ -0,0 +1,136 @@
+package com.epam.tr.service;
+
+import com.epam.tr.dto.FileDto;
+import com.epam.tr.dto.FileRequestDto;
+import com.epam.tr.service.logic.builder.FileEntityBuilder;
+import com.epam.tr.service.logic.builder.PathBuilder;
+import com.epam.tr.service.logic.sorter.SorterFactory;
+import com.epam.tr.service.logic.validator.Validator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.epam.tr.dto.FileDto.FileSystemObjectType.*;
+import static java.io.File.listRoots;
+import static java.util.Arrays.stream;
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class FileSystemService implements FileService {
+
+    @Autowired
+    private Validator<FileDto> validator;
+    @Autowired
+    private FileEntityBuilder builder;
+    @Autowired
+    private PathBuilder pathBuilder;
+
+    @Override
+    public void create(FileRequestDto requestDto) throws IOException {
+        FileDto fileDto = builder.buildFileEntity(requestDto);
+        validator.isValid(fileDto);
+        if (fileDto.getType() == FOLDER) {
+            createFolder(fileDto);
+        }
+        if (fileDto.getType() == FILE) {
+            createFile(fileDto);
+        }
+    }
+
+    @Override
+    public void delete(FileRequestDto requestDto) throws IOException {
+        FileDto fileDto = builder.buildFileEntity(requestDto);
```

Comment:
Actually this is super strange.
You are processing a dto in buildFileEntity then validating... but validating should be first, best way to validate DTO it is and annotation Valid to the endpoint and and validation annotations to DTO's fields.